engine/dlib/src/dlib/buffer.cpp

```diff
@@ -616,4 +644,117 @@ namespace dmBuffer
+        dmArray<Buffer::MetaData*>& arr = buffer->m_MetaDataArray;
+        for (uint32_t  i=0; i<arr.Size(); i++) {
+            if (arr[i]->m_Name == name_hash) {
+                return arr[i];
+            }
+        }
+        return 0; // nothing found
+    }
+
+
+    Result GetMetaData(HBuffer hbuffer, dmhash_t name_hash, void** data, uint32_t* count, ValueType* type) {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+
+        Buffer::MetaData* item = FindMetaDataItem(buffer, name_hash);
+        if (!item) {
+            return RESULT_METADATA_NOT_EXIST;
+        }
+
+        *count = item->m_ValueCount;
+        *type = (ValueType)item->m_ValueType; // assumes value tpe is sanitized when setting the metadata
+        *data = item->m_Data;
+
+        return RESULT_OK;
+    }
+
+    Result SetMetaData(HBuffer hbuffer, dmhash_t name_hash, const void* data, uint32_t count, ValueType type) {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+        if (count == 0) {
+            return RESULT_METADATA_INVALID;
+        }
+
+        Buffer::MetaData* item = FindMetaDataItem(buffer, name_hash);
+        uint32_t values_block_size = GetSizeForValueType(type)*count; // do this once
+        if (item) {
+            // make sure the type and value count is right
+            if (item->m_ValueCount != count || item->m_ValueType != type) {
+                return RESULT_METADATA_INVALID; // TODO use a proper error status hsere
+            }
+            memcpy(item->m_Data, data, values_block_size);
+        } else {
+            dmArray<Buffer::MetaData*>& metadata_items = buffer->m_MetaDataArray;
+            if (metadata_items.Remaining() <= 0) {
+                metadata_items.OffsetCapacity(2); // TODO - shall we set an upper limit to the number of meta data entries ?
+            }
+            if (metadata_items.Remaining() > 0) {
+                item = (Buffer::MetaData*) malloc(sizeof(Buffer::MetaData));
+                item->m_Name = name_hash;
+                item->m_ValueCount = count; // no restriction in max number of values
+                item->m_ValueType = type; // no validation, all ValueTypes do
+                item->m_Data = malloc(values_block_size);
+                memcpy(item->m_Data, data, values_block_size);
+                metadata_items.Push(item);
+            } else {
+                return RESULT_ALLOCATION_ERROR;
+            }
+        }
+
+        return RESULT_OK;
+    }
+
+    Result SetBounds(HBuffer hbuffer, const dmVMath::Point3& min, const dmVMath::Point3& max) {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+
+        Result r = SetMetaData(hbuffer, dmHashString64("min_aabb"), &min, 3, VALUE_TYPE_FLOAT32);
```

Comment:
Instead of donig these hashes each time the function is called, I recommend adding a constant a the file scope. E.g. `dmBuffer:BUFFER_AABB_HASH = dmHashString("aabb");`

Reply:
No need to fix. SetBounds() will be removed.